src/cljs/org/broadinstitute/firecloud_ui/page/workspace/submission_details.cljs

```diff
@@ -145,12 +176,12 @@
                             :verticalAlign "middle"}}
              (submission "status")]]
            (when (= "Submitted" (submission "status"))
-             [:div {:style {:padding "0.7em 0" :marginTop "0.5em" :color "#fff" :cursor "pointer"
-                            :backgroundColor (:exception-red style/colors) :borderRadius 5}
-                    :onClick #(js/console.log "Insert abort logic here")}
-              (icons/font-icon {:style {:fontSize "160%" :marginRight 14 :verticalAlign "middle"}}
-                :status-warning-triangle)
-              [:span {:style {:fontSize "125%" :verticalAlign "middle"}} "Abort"]])]
+             [AbortButton
+              {:on-abort (fn []
+                          (swap! state assoc :server-response nil)
+                          (react/call :load-details this))
+               :workspace-id (:workspace-id props)
+               :submission submission}])]
```

Comment:
I think you can get by with passing only the submission ID rather than the entire submission object/map.